src/main/java/com/aws/greengrass/util/platforms/unix/linux/CGroupV2.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms.unix.linux;
+
+import com.aws.greengrass.lifecyclemanager.GreengrassService;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Set;
+
+@SuppressFBWarnings(value = "DMI_HARDCODED_ABSOLUTE_FILENAME",
+        justification = "CGroupV2 virtual filesystem path cannot be relative")
+public enum CGroupV2 implements CGroupSubSystemPaths {
+    Memory, CPU, Freezer, Unified;
+    private static final String CGROUP_SUBTREE_CONTROL_CONTENT = "+cpuset +cpu +io +memory +pids";
+
+    @Override
+    public String rootMountCmd() {
+        return String.format("mount -t cgroup2 none %s", CGROUP_ROOT);
+    }
+
+    @Override
+    public Path getSubsystemRootPath() {
+        return CGROUP_ROOT;
+    }
+
+    @Override
+    public Path getComponentMemoryLimitPath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(MEMORY_MAX);
+    }
+
+    @Override
+    public Path getCgroupFreezerStateFilePath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(CGROUP_FREEZE);
+    }
+
+    @Override
+    public Path getRootSubTreeControlPath() {
+        return getSubsystemRootPath().resolve(CGROUP_SUBTREE_CONTROL);
+    }
+
+    @Override
+    public Path getGGSubTreeControlPath() {
+        return getSubsystemGGPath().resolve(CGROUP_SUBTREE_CONTROL);
+    }
+
+    @Override
+    public Path getComponentCpuMaxPath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(CPU_MAX);
+    }
+
+    @Override
+    public Path getCgroupFreezePath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(CGROUP_FREEZE);
+    }
+
+    @Override
+    public void initializeCgroup(GreengrassService component, LinuxPlatform platform) throws IOException {
```

Comment:
how does this method differ from what is in CgroupV1? Can they share the same implementation? This code looks like it is just duplicated.